### ISSUE TITLE
Colorized cli output is impossible to read

### ISSUE THREAD

**Issue by omajid** [dotnet/efcore#9766]
The `dotnet ef` cli tool colourizes output to make it easier to read. In some cases, it actually becomes impossible to read.

Here is what I see when I run `dotnet ef migrations list` on my machine:

![Terminal screenshot](https://user-images.githubusercontent.com/174451/30296868-b5db828a-9713-11e7-9867-4ba4d0c96dd8.png)

I would expect the output to be always be readable, no matter the colorscheme in my terminal. In fact, I would rather have colourless output than something unreadable.

### Steps to reproduce

- Use a recent version of GNOME Terminal on Linux with the *Solarized Dark* colorscheme.
- `dotnet new web`
- Add the following to the csproj file:
```
    <PackageReference Include="Microsoft.EntityFrameworkCore.SQLite" Version="2.0.0" />
    <DotNetCliToolReference Include="Microsoft.EntityFrameworkCore.Tools.DotNet" Version="2.0.0" />
```
- Create a `Model.cs` file. Use https://docs.microsoft.com/en-us/ef/core/get-started/netcore/new-db-sqlite#create-the-model but introduce a typo.
- `dotnet ef migrations list`
- `dotnet ef migrations list --verbose`

### Further technical details
EF Core version:
```
    <PackageReference Include="Microsoft.EntityFrameworkCore.SQLite" Version="2.0.0" />                              
    <DotNetCliToolReference Include="Microsoft.EntityFrameworkCore.Tools.DotNet" Version="2.0.0" />
```

Database Provider: `Microsoft.EntityFrameworkCore.SQLite`

Operating system:

    $ uname -a
    Linux trull 4.12.9-300.fc26.x86_64 #1 SMP Fri Aug 25 13:09:43 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
    $ cat /etc/os-release 
    NAME=Fedora
    VERSION="26 (Workstation Edition)"
    ID=fedora
    VERSION_ID=26
    PRETTY_NAME="Fedora 26 (Workstation Edition)"
    ANSI_COLOR="0;34"
    CPE_NAME="cpe:/o:fedoraproject:fedora:26"
    HOME_URL="https://fedoraproject.org/"
    BUG_REPORT_URL="https://bugzilla.redhat.com/"
    REDHAT_BUGZILLA_PRODUCT="Fedora"
    REDHAT_BUGZILLA_PRODUCT_VERSION=26
    REDHAT_SUPPORT_PRODUCT="Fedora"
    REDHAT_SUPPORT_PRODUCT_VERSION=26
    PRIVACY_POLICY_URL=https://fedoraproject.org/wiki/Legal:PrivacyPolicy
    VARIANT="Workstation Edition"
    VARIANT_ID=workstation

IDE: cli:

    $ dotnet --info
    .NET Command Line Tools (2.0.0)
    
    Product Information:
     Version:            2.0.0
     Commit SHA-1 hash:  cdcd1928c9
    
    Runtime Environment:
     OS Name:     fedora
     OS Version:  26
     OS Platform: Linux
     RID:         fedora.26-x64
     Base Path:   /usr/lib64/dotnet/sdk/2.0.0/
    
    Microsoft .NET Core Shared Framework Host
    
      Version  : 2.0.0
      Build    : N/A

**Comment by bricelam** [comment]
Do you have bold disabled?

Try the following, and if some of the colors look the same, you may want to file a new issue on [dotnet/corefx](https://github.com/dotnet/corefx).

``` csharp
Console.ForegroundColor = ConsoleColor.Black;
Console.WriteLine("Black");
Console.ForegroundColor = ConsoleColor.DarkGray;
Console.WriteLine("DarkGray");
Console.ForegroundColor = ConsoleColor.Gray;
Console.WriteLine("Gray");
Console.ForegroundColor = ConsoleColor.White;
Console.WriteLine("White");
```

It should look the same as doing this in the terminal.

``` sh
echo -e "\e[30mBlack\e[0m"
echo -e "\e[30;1mDarkGray\e[0m"
echo -e "\e[37mGray\e[0m"
echo -e "\e[37;1mWhite\e[0m"
```

We print verbose messages using `DarkGray`. It may also be the case that this is conflicting with your current theme.

Workaround
===
You can always workaround coloring issues by using `--no-color`. Output semantics can be preserved by also adding `--prefix-output`

**Comment by omajid** [comment]
> Do you have bold disabled?

I dont think so.

> Try the following

Here is what I see: ![screenshot from 2017-09-13 17-22-34](https://user-images.githubusercontent.com/174451/30401485-6bc7da86-98a8-11e7-99e3-71d892653e14.png)

> You can always workaround coloring issues by using --no-color

That works, even if it is a little cumbersome. Thanks.

**Comment by omajid** [comment]
Would it be possible to keep this issue open? There is a workaround, sure. But the underlying issue is still a bug and makes it harder for me to use the EF cli tooling.

**Comment by bricelam** [comment]
We believe this is an issue with the theme: Dark gray (a.k.a. bold black) isn't visible.

Other utilities (like MSBuild) also output verbose messages in this color, so it seems external to EF. If there's a well-established pattern to ensure applications never output to color that would be invisible in the current theme, we're certainly open to taking a PR for this. Last I checked, however, .NET Core didn't implement the `Console.BackgroundColor` getter in a way that would be useful in this scenario, so we might need an issue in [dotnet/corefx](https://github.com/dotnet/corefx) to unblock this work.

**Comment by BrennanConroy** [comment]
Why not do what the Console logger does? https://github.com/aspnet/Logging/blob/671af986ec3b46dc81e28e4a6c37a9d0ee283c65/src/Microsoft.Extensions.Logging.Console/ConsoleLogger.cs#L234
